src/net.md

```diff
@@ -1019,6 +1020,93 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-progress-with-range]: #ex-progress-with-range
+<a name="ex-progress-with-range"/>
+## Make a partial download with HTTP range headers
+
+[![reqwest-badge]][reqwest] [![cat-net-badge]][cat-net]
+
+Uses [`reqwest::Client::head`] to get the content-length and validate if the server sets the header
+[`reqwest::header::ContentRange`], required to confirm the support of partial downloads.
+
+If supported downloads the content using [`reqwest::get`], setting the [`reqwest::header::Range`]
+to do partial downloads in chunks of 100 bytes, between these writes basic progress messages.
+
+Range header, defined in [RFC7233][HTTP Range RFC7233].
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+extern crate reqwest;
+
+use std::io::Read;
+
+use reqwest::header::{Range, ContentRange, ContentRangeSpec};
+
+# error_chain! {
+#     foreign_links {
+#         Io(std::io::Error);
+#         Reqwest(reqwest::Error);
+#     }
+# }
+
+fn run() -> Result<()> {
+    let client = reqwest::Client::new();
+
+    let url = "https://httpbin.org/range/1024?duration=2";
+    let response = client.head(url).send()?;
+
+    let range = response.headers().get::<ContentRange>().ok_or(
+        "response doesn't include the expected ranges",
+    )?;
+
+    let content_length = match **range {
+        ContentRangeSpec::Bytes { instance_length: Some(len), .. } => len,
+        _ => bail!("response doesn't include the expected ranges"),
+    };
+
+    let mut content: Vec<u8> = Vec::with_capacity(content_length as usize);
+    println!("content-lenght: {} download in progress..", content_length);
+
+    let mut chunk_starts: u64 = 0;
+    let mut chunk_ends = if 100 > content_length {
```

Comment:
Although I would suggest not to hardcode "100" here but have some value/constant indicating the buffer size

Also are you upto converting the offset calculation code into an iterator returning `Range::bytes`? That would be awesome!

Reply:
I can try that one, although looks like in the `std` `Range::step_by()` is still in the nightly build only [#27741](https://github.com/rust-lang/rust/issues/27741). I'll do it for now manually inside the for loop.

Reply:
the `step_by` would not help here much as it would just return every N'th value of the range (treated as iterator here) so
```rust
 for s in (0..1024).step_by(100) {
    println!("{:?}", s);
 }
```
would give 
```
0
100
200
300
400
500
600
700
800
900
1000
```

Which would help with the implementation but would not give a full solution. 
I would propose to make own iterator returning 
`hyper::header::Range::bytes` with each iteration containing the next range we want to obtain

Reply:
Anyhow, if you'll have any problems or concerns I'll gladly help!

Reply:
@budziq please let me know if it's ok, and if you wanna hide some of these extra lines. Thanks!